scripts/autotools/libmongocrypt/CheckSSL.m4

```diff
@@ -0,0 +1,35 @@
+if test "$PHP_CLIENT_SIDE_ENCRYPTION" != "no"; then
+  AC_MSG_CHECKING([which crypto library to use for libmongocrypt])
+  AC_MSG_RESULT([$PHP_MONGODB_SSL])
+
+  dnl Disable Windows crypto
+  AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_CNG, 0)
+
+  if test "$PHP_MONGODB_SSL" = "darwin"; then
+    AC_SUBST(MONGOC_ENABLE_CLIENT_SIDE_ENCRYPTION, 1)
```

Comment:
I'd prefer to keep this alongside the other `AC_SUBST` calls for `MONGOC_` constants, even if it means an extra `"$PHP_CLIENT_SIDE_ENCRYPTION" != "no"` conditional.

One thing that may help is if you look in libmongoc's own `CheckSSL.m4`, you'll see that we re-assign `$PHP_MONGODB_SSL` once we know that we've selected a TLS library. If we cannot (for the auto case), we'll ultimately assign it to "no", such that the variable is only either "yes" or "no" after that include is run.

If you were to do that in this file, then we can have a definitive yes/no value for `$PHP_CLIENT_SIDE_ENCRYPTION` after all of our m4 includes when the time comes to do our `AC_SUBST`, `PHP_MONGODB_ADD_SOURCES`, and other calls from `config.m4`.

Reply:
CheckSSL.m4 now re-assigns `PHP_CLIENT_SIDE_ENCRYPTION`, while an additional check in config.m4 ensures that `MONGOC_ENABLE_CLIENT_SIDE_ENCRYPTION` is set before the first `AC_CONFIG_FILES` for libmongoc. A second `PHP_CLIENT_SIDE_ENCRYPTION` conditional then adds sources for compiling with libmongocrypt to keep logic separated and grouped where appropriate.